backend/src/services/advancedHolderAnalysisService.js

```
@@ -0,0 +1,849 @@
+        
+        // Calculate percentages
+        Object.keys(distribution).forEach(tier => {
+            distribution[tier].percentage_of_supply = 
+                (distribution[tier].total_tokens / this.TOTAL_TOKEN_SUPPLY) * 100;
+            
+            // Sort holders within tier by token amount
+            distribution[tier].holders.sort((a, b) => b.tokens - a.tokens);
+            
+            // Keep only top 10 per tier for API response
+            distribution[tier].top_holders = distribution[tier].holders.slice(0, 10);
+            delete distribution[tier].holders; // Remove full list to reduce response size
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)